arm9/src/unk_02006D98.c

```diff
@@ -0,0 +1,2488 @@
+            param0->unk5C[*r6]++;
+
+            if (param0->unk84[*r6].unk1 == param0->unk5C[*r6] || param0->unk84[*r6].unk1 == 0)
+            {
+                param0->unk5C[*r6] = 0;
+                r6++;
+            }
+            else
+            {
+                *r6 = -2 - param0->unk84[*r6].unk0;
+            }
+        }
+
+        if (param0->unk84[*r6].unk0 == -1 || *r6 >= 10)
+        {
+            param0->unk5B = 0;
+            param0->unk58 = 0;
+
+            return;
+        }
+
+        param0->unk5B = param0->unk84[*r6].unk0;
+        param0->unk5A = param0->unk84[*r6].unk1;
+    }
+    else
+    {
+        param0->unk5A--;
+    }
+}
+#else
+asm void FUN_02007F48(struct UnkStruct_02006D98_2 *param0)
+{
+    // clang-format off
+	push {r4-r7}
+	add r1, r0, #0x0
+	add r1, #0x58
+	ldrb r1, [r1, #0x0]
+	add r6, r0, #0x0
+	add r6, #0x59
+	cmp r1, #0x0
+	beq _0200800A
+	add r1, r0, #0x0
+	add r1, #0x5a
+	ldrb r1, [r1, #0x0]
+	cmp r1, #0x0
+	bne _02007FFE
+	ldrb r1, [r6, #0x0]
+	add r1, r1, #0x1
+	strb r1, [r6, #0x0]
+	ldrb r3, [r6, #0x0]
+	mov r1, #0x84
+	lsl r4, r3, #0x1
+	add r2, r0, r4
+	ldrsb r2, [r2, r1]
+	sub r1, #0x85
+	cmp r2, r1
+	bge _02007FC6
+	mov r2, #0x0
+	sub r1, r2, #0x2
+_02007F7C:
+	add r4, r0, r3
+	add r4, #0x5c
+	ldrb r4, [r4, #0x0]
+	add r3, r0, r3
+	add r3, #0x5c
+	add r4, r4, #0x1
+	strb r4, [r3, #0x0]
+	ldrb r5, [r6, #0x0]
+	lsl r3, r5, #0x1
+	add r4, r0, r3
+	add r3, r4, #0x0
+	add r7, r0, r5
+	add r3, #0x85
+	add r7, #0x5c
+	ldrb r3, [r3, #0x0]
+	ldrb r7, [r7, #0x0]
+	cmp r3, r7
+	beq _02007FA4
+	cmp r3, #0x0
+	bne _02007FAE
+_02007FA4:
+	add r3, r0, r5
+	add r3, #0x5c
+	strb r2, [r3, #0x0]
+	add r6, r6, #0x1
+	b _02007FB6
+_02007FAE:
+	mov r3, #0x84
+	ldrsb r3, [r4, r3]
+	sub r3, r1, r3
+	strb r3, [r6, #0x0]
+_02007FB6:
+	ldrb r3, [r6, #0x0]
+	mov r5, #0x84
+	lsl r4, r3, #0x1
+	add r7, r0, r4
+	ldrsb r7, [r7, r5]
+	sub r5, #0x85
+	cmp r7, r5
+	blt _02007F7C
+_02007FC6:
+	add r2, r0, r4
+	mov r1, #0x84
+	ldrsb r2, [r2, r1]
+	sub r1, #0x85
+	cmp r2, r1
+	beq _02007FD6
+	cmp r3, #0xa
+	blo _02007FE6
+_02007FD6:
+	add r1, r0, #0x0
+	mov r2, #0x0
+	add r1, #0x5b
+	strb r2, [r1, #0x0]
+	add r0, #0x58
+	strb r2, [r0, #0x0]
+	pop {r4-r7}
+	bx lr
+_02007FE6:
+	add r1, r0, #0x0
+	add r1, #0x5b
+	strb r2, [r1, #0x0]
+	ldrb r1, [r6, #0x0]
+	lsl r1, r1, #0x1
+	add r1, r0, r1
+	add r1, #0x85
+	ldrb r1, [r1, #0x0]
+	add r0, #0x5a
+	strb r1, [r0, #0x0]
+	pop {r4-r7}
+	bx lr
+_02007FFE:
+	add r1, r0, #0x0
+	add r1, #0x5a
+	ldrb r1, [r1, #0x0]
+	add r0, #0x5a
+	sub r1, r1, #0x1
+	strb r1, [r0, #0x0]
+_0200800A:
+	pop {r4-r7}
+	bx lr
+    // clang-format on
+}
+#endif
+
+THUMB_FUNC void FUN_02008010(u8 *param0, struct UnkStruct_02006D98_sub *param1)
+{
+    param0[0] = 1;
+    param0[2] = 0;
+    param0[1] = (u8)param1->unk0;
+    param0[3] = param1->unk1;
+    ((struct UnkStruct_02006D98_sub **)param0)[4] = param1;
+
+    for (s32 i = 0; i < 10; i++)
+    {
+        param0[i + 4] = 0;
+    }
+}
+
+#ifdef NONMATCHING
```

Comment:
what's the nonmatching behavior here?

Reply:
Mainly register allocation issues. Both problematic functions are similar, maybe there is a macro/inline here too